### PR TITLE
feat: add IsFirebasePushId validator

### PR DESCRIPTION
**Pull request by elmarti**
This PR adds support for Firebase-style PushIds, you can see more about the science and inner-workings here: https://firebase.googleblog.com/2015/02/the-2120-ways-to-ensure-unique_68.html

**Overview**
Simply, we're checking:
- The input must be a string
- The string must be 20 characters in length
- The string must only contain web-safe characters

I've checked the backlog and PRs for duplicates and found nothing.

### CODE REVIEW

File: test/functional/reject-validation.spec.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import "es6-shim";
+import "es6-shim";
```
Comment (vlapo) [inline test/functional/reject-validation.spec.ts]: duplicate row

File: README.md [hunk 2]
```diff
@@ -951,6 +952,7 @@ validator.isInstance(value, target); // Checks value is an instance of the targe
 | `@IsSurrogatePair()`                            | Checks if the string contains any surrogate pairs chars.                                                                         |
 | `@IsUrl(options?: IsURLOptions)`                | Checks if the string is an url.                                                                                                  |
 | `@IsUUID(version?: "3"\|"4"\|"5"\|"all")`              | Checks if the string is a UUID (version 3, 4, 5 or all ).                                                                              |
+| `@IsPushId()`                                   | Checks if the string is a Firebase Push id                                                                                      |
```
Comment (vlapo) [inline README.md]: add link to firebase PushId specification

**Comment by vlapo** [comment]
Also name `IsPushId` is really misleading. What about `IsFirebasePushId` ?

**Comment by elmarti** [comment]
@vlapo Agreed, although the Firebase part was excluded as they can exist and be used apart from Firebase, however in practice they're not really.

**Pull request merged**